northstar/src/runtime/process/init.rs

```diff
@@ -76,34 +100,21 @@ impl Init {
         // Clone
         match unsafe { unistd::fork() } {
             Ok(result) => match result {
-                unistd::ForkResult::Parent { child } => {
+                unistd::ForkResult::Parent { .. } => {
                     // Drop checkpoint. The fds are cloned into the child and are closed upon execve.
                     drop(condition_notify);
 
-                    // Wait for the child to exit
-                    loop {
-                        match waitpid(Some(child), None) {
-                            Ok(WaitStatus::Exited(_pid, status)) => {
-                                let exit_status = ExitStatus::Exit(status);
-                                exit_status_channel
-                                    .send(&exit_status)
-                                    .expect("Failed to send exit status");
-                                exit(0);
-                            }
-                            Ok(WaitStatus::Signaled(_pid, status, _)) => {
-                                let exit_status = ExitStatus::Signalled(status as u8);
-                                exit_status_channel
-                                    .send(&exit_status)
-                                    .expect("Failed to send exit status");
-                                exit(0);
-                            }
-                            Ok(WaitStatus::Continued(_)) | Ok(WaitStatus::Stopped(_, _)) => {
-                                continue
-                            }
-                            Err(nix::Error::EINTR) => continue,
-                            e => panic!("Failed to waitpid on {}: {:?}", child, e),
-                        }
-                    }
+                    // FIXME: this is a workaround to be able to have a multi-threaded runtime
+                    // running inside the init process. It should be possible to remove this
+                    // once we implement have a separate process that does the forking.
+                    std::thread::spawn(move || {
+                        let rt = tokio::runtime::Builder::new_multi_thread().build().unwrap();
```

Comment:
Is the MT rt needed to handle blocking stuff e.g `waitpid` in future? Currently a `current_thread` runtime would be cheaper and sufficient.

If we need a MT RT, we should probably fine tune the number of [worker_threads](https://docs.rs/tokio/1.14.0/tokio/runtime/struct.Builder.html#method.worker_threads).

Reply:
yeah, this is true, this was just an experiment :sweat_smile: i'll use a single thread and see if we can get away with that